tremor-script/src/srs.rs

```diff
@@ -503,11 +503,18 @@ impl EventPayload {
     ///   });
     ///   println!("v: {}", v)
     /// ```
-    pub fn rent<F, R>(&self, f: F) -> R
+    pub fn rent<'iref, F, R>(&'iref self, f: F) -> R
     where
-        F: for<'iref, 'head> FnOnce(&'iref ValueAndMeta<'head>) -> R,
+        F: for<'head> FnOnce(&'head ValueAndMeta<'head>) -> R,
     {
-        f(&self.data)
+        // ALLOW: we are turning a longer lifetime into a shorter one for a
+        // covariant type ValueAndMeta. &mut is invariant over it's lifetime,
+        // but we are choosing a shorter one and passing it down not up. So a
+        // user should not be able to choose an inappropriate lifetime for it,
+        // plus they don't control the owner here.
+        f(unsafe {
+            mem::transmute::<&'iref ValueAndMeta<'static>, &'iref ValueAndMeta<'iref>>(&self.data)
```

Comment:
The allow checks are pretty basic, using grep not parsing or clippy
```suggestion
        // we are turning a longer lifetime into a shorter one for a covariant
        // type ValueAndMeta. &mut is invariant over it's lifetime, but we are
        // choosing a shorter one and passing it down not up. So a user
        // should not be able to choose an inappropriate lifetime for it, plus
        // they don't control the owner here.
        f(unsafe {
            // ALLOW: See above explenation
            mem::transmute::<&'iref ValueAndMeta<'static>, &'iref ValueAndMeta<'iref>>(&self.data)
```